tests/components/template/test_sensor.py

```diff
@@ -438,6 +442,36 @@ def test_available_template_with_entities(self):
         )
 
 
+async def test_creating_sensor_loads_group(hass):
+    """Test setting up template sensor loads group component first."""
+    order = []
```

Comment:
@Tho85 

Not sure if it will work, but .. maybe

`hass.bus.async_listen(EVENT_COMPONENT_LOADED, async_add_to_order)`

Reply:
It didn't, but because I was doing the wrong thing.

The sensor _component_ loads before the group component in any case, but the sensor.template _platform_ waits until the group component has finished initialization. I fixed my test by patching the `async_setup_platform` method for sensor.template, instead of `async_setup` for sensor.

Now the test should be reliable :tada: